internal/addon/filter.go

```diff
@@ -0,0 +1,23 @@
+package addon
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetResourcesByLabelSelector returns the list of resources of type T matching the label selector.
+// It works generically for any Kubernetes resource that implements client.Object.
+func GetResourcesByLabelSelector[T client.Object](resources []client.Object, selector map[string]string) []T {
```

Comment:
WDYT of this rename?
```suggestion
func FilterResourcesByLabelSelector[T client.Object](resources []client.Object, selector map[string]string) []T {
```